src/components/CortexGroupActionItemsWidget/CortexGroupActionItemsWidget.test.tsx

```diff
@@ -0,0 +1,167 @@
+ */
+
+import { CortexApi } from '../../api/CortexApi';
+import React from 'react';
+import {
+  act,
+  fireEvent,
+  render,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+import { TestApiProvider, wrapInTestApp } from '@backstage/test-utils';
+import { cortexApiRef } from '../../api';
+import {
+  EntityProvider,
+  entityRouteRef,
+} from '@backstage/plugin-catalog-react';
+import { initiativeRouteRef, rootRouteRef } from '../../routes';
+import { CortexGroupActionItemsWidget } from './CortexGroupActionItemsWidget';
+
+describe('<CortexGroupActionItemsWidget/>', () => {
+  const entity = {
+    apiVersion: 'v1',
+    kind: 'Group',
+    metadata: {
+      name: 'ExampleGroup',
+    },
+    spec: {
+      type: 'group',
+    },
+  };
+
+  const cortexApi: Partial<CortexApi> = {
+    getInitiativeActionItemsForTeam: () =>
+      Promise.resolve([
+        {
+          rule: {
+            id: 1,
+            expression: 'git != null',
+            weight: 1,
+          },
+          componentRef: 'component1',
+          initiative: {
+            initiativeId: 1,
+            name: 'Basic service catalog',
+            targetDate: '01/01/2000',
+          },
+        },
+        {
+          rule: {
+            id: 1,
+            expression: 'git != null',
+            weight: 1,
+          },
+          componentRef: 'component1',
+          initiative: {
+            initiativeId: 2,
+            name: 'Git initiative',
+            targetDate: '05/05/2000',
+          },
+        },
+        {
+          rule: {
+            id: 2,
+            expression: 'description != null',
+            weight: 1,
+          },
+          componentRef: 'component1',
+          initiative: {
+            initiativeId: 1,
+            name: 'Basic service catalog',
+            targetDate: '01/01/2000',
+          },
+        },
+        {
+          rule: {
+            id: 2,
+            expression: 'description != null',
+            weight: 1,
+          },
+          componentRef: 'component2',
+          initiative: {
+            initiativeId: 1,
+            name: 'Basic service catalog',
+            targetDate: '01/01/2000',
+          },
+        },
+      ]),
+  };
+
+  const renderWrapped = (children: React.ReactNode) =>
+    render(
+      wrapInTestApp(
+        <TestApiProvider apis={[[cortexApiRef, cortexApi]]}>
+          <EntityProvider entity={entity}>{children}</EntityProvider>
+        </TestApiProvider>,
+        {
+          mountedRoutes: {
+            '/': rootRouteRef,
+            '/entity/:namespace/:kind/:name': entityRouteRef,
+            '/initiatives/:id': initiativeRouteRef as any,
+          },
+        },
+      ),
+    );
+
+  it('Properly renders CortexGroupActionItemsWidget', async () => {
+    const { findByText, queryByText, queryAllByText, queryByLabelText } =
+      renderWrapped(<CortexGroupActionItemsWidget />);
+    expect(await findByText(/component1/)).toBeInTheDocument();
```

Comment:
I wonder if it's worth starting a TestUtils file, similar to what we have in Cortex that does like: 

checkForText = (text: string) => expect(await findByText(/component1/)).toBeInTheDocument()

and so on

Reply:
Discussed on Slack and we won't set up the util file. The existing tests are pulling the testing library functions from the object being returned by `render()`. Using `screen` causes the test to fail. For example, this works:

```
    const { findByText } = renderWrapped(<CortexScorecardWidget />);
    expect(await findByText(/42%/)).toBeInTheDocument();
```

This fails:
```
expect(screen.queryAllByText(matcher)[index ?? 0]).toBeVisible();
```

Thus, a test util file would need to pass in the `RenderResult` from `render()` in order for stuff to work, which is pretty ugly (see below), so we won't do it.

```
export function checkForText2(renderResult: RenderResult, matcher: Matcher, index?: number) {
  const { queryAllByText } = renderResult;
  expect(queryAllByText(matcher)[index ?? 0]).toBeVisible();
}
```

Reply:
don't wanna block but we can have a TestUtil that wraps `render` (`renderTestUtil` or something) and augments it with helper functions, like `checkForText`, etc -- just to keep in mind for later PRs